lib/plugins/mastodon.py

```diff
@@ -0,0 +1,62 @@
+import os
+import textwrap
+
+import requests
+from bs4 import BeautifulSoup
+from mastodon import Mastodon
+
+
+class mastodon_client:
+    def __init__(self, **kwargs):
+        self.base_url = kwargs.get("base_url", "https://mstdn.science")
+        self.mastodon_handle = Mastodon(
+            access_token=kwargs.get("access_token"), api_base_url=self.base_url
+        )
+        self.max_content_length = kwargs.get("max_content_length", 500)
+
+    def create_post(self, content, mentions, hashtags, images):
+        media_ids = []
+        for image in images[:4]:
+            response = requests.get(image["url"])
+            filename = image["url"].split("/")[-1]
+            if response.status_code == 200:
+                with open(filename, "wb") as f:
+                    f.write(response.content)
+                media_uploaded = self.mastodon_handle.media_post(
+                    media_file=filename,
+                    description=image["alt_text"] if "alt_text" in image else None,
+                )
+                media_ids.append(media_uploaded["id"])
+                os.remove(filename)
```

Comment:
Can you use a proper temporary file here, for which you ensure that it's removed under all circumstances?